lib/x86_cpu/x86_env.ml

```diff
@@ -123,6 +124,7 @@ module type ModeVars = sig
   val r : var array
   val nums : var array
   val ymms : var array
+  val of_reg : reg -> var option
```

Comment:
Changing the module type is a breaking change since we're changing an abstraction, which is contravariant, as module types are used as parameters to functors. 

Moreover, the `of_reg` function is LLVM specific, and we are hiding this. For example, it relies on a `decode` function, that presumes the llvm encoding, and fails with the `Failure` exception if it is not the case. So my suggestion would be to factor it out in a separate module, which has an explicit dependency on llvm. (We don't really need the build time dependency, as I'm envisioning cases, when we need to analyze LLVM output, without actually using LLVM library, e.g., when a serialized version of a project is passed to another machine). But we still need to make it clear, both in documentation and in the choice of names, that this mapping is LLVM specific, e.g., `bap_x86_llvm_cpu_env.ml`.

Reply:
happy to change it. however, this doesn't break semantic versioning does it? the previous interface remains intact I thought.

Reply:
it breaks, it changes the module type of ModeVars, thus any function/functor that were taking ModeVars as a parameter will break. 

Basically, semantic version treats the subtyping of a library interface as a non-breaking change. So we can make the interface more specific by adding new things. But as always with subtyping 
we need to distinguish between covariant and contravariant types. So we may add new stuff to covariant types, and erase fields from contravariant. We can't touch invariant types. In our case, ModeVars is invariant, so we can't add or remove anything to it without breaking code.